app/routes.py

```diff
@@ -1,2 +1,277 @@
+
+        else:
+            return (f"None", 404)
+    elif request.method == "PUT":
+        if task:
+            request_body = request.get_json()
+            task.title = request_body["title"]
+            task.description = request_body["description"]
+            task.completed_at = request_body["completed_at"]
+            db.session.commit()
+            if task.completed_at == None:
+                is_complete = False 
+            else:
+                is_complete = True 
+            
+            return {
+                "task":{
+                    "id": task.task_id,
+                    "title": task.title,
+                    "description": task.description,
+                    "is_complete": is_complete
+            }}, 200
+
+        else:
+            return (f"None", 404)
+
+    elif request.method == "DELETE":
+        if task:
+            db.session.delete(task)
+            db.session.commit()
+            return {
+                "details": f"Task {task.task_id} \"{task.title}\" successfully deleted"
+            }, 200
+        else:
+            return (f"None", 404)
+
+    else:
+        return (f"None", 404)
+
+@tasks_bp.route("/<task_id>/mark_complete", methods=["PATCH"], strict_slashes=False)
+def mark_complete(task_id):
+    task = Task.query.get(task_id)
+    if not task:
+        return "", 404
+
+    task.completed_at = date.today()
+    db.session.commit()
+    call_slack(task)
+    return {
+        "task": {
+        "id": task.task_id,
+        "title": task.title,
+        "description": task.description,
+        "is_complete": True
+}}, 200
+
+
+@tasks_bp.route("/<task_id>/mark_incomplete", methods=["PATCH"], strict_slashes=False)
+def mark_incomplete(task_id):
+    task = Task.query.get(task_id)
+    if task:
+        task.completed_at = None
+        db.session.commit()
+        return {
+            "task": {
+            "id": task.task_id,
+            "title": task.title,
+            "description": task.description,
+            "is_complete": False
+    }}, 200
+
+    else:
+        return (f"None", 404)
+
+def call_slack(task):
+    key = os.environ.get("api_key")
+    url = "https://slack.com/api/chat.postMessage"
+    slack_str = f"Someone just completed the task {task.title}"
+    requests.post(url, data = {"token": key, "channel": "building-a-robot", "text": slack_str})
+
+@goals_bp.route("", methods=["POST", "GET"], strict_slashes=False)
+def handle_goals():
+    if request.method == "GET":  
+        query_param_value = request.args.get("sort")
+        if query_param_value == "asc":
+            goals = Goal.query.order_by(Goal.title.asc())
+
+        elif query_param_value == "desc":
+            goals = Goal.query.order_by(Goal.title.desc())
+
+        else:
+            goals = Goal.query.all()
+        goals_response = []
+        for goal in goals:
+            goals_response.append({
+                "id": goal.goal_id,
+                "title": goal.title
```

Comment:
This is fine as is but since you defined those `to_json` and `to_dict` methods on the Goal class, this would be a good place to call one of them!